src/coreclr/jit/importer.cpp

```diff
@@ -3970,13 +3969,9 @@ GenTree* Compiler::impImportStaticFieldAddress(CORINFO_RESOLVED_TOKEN* pResolved
             op1         = gtNewIconHandleNode(fldAddr, handleKind, innerFldSeq);
             INDEBUG(op1->AsIntCon()->gtTargetHandle = reinterpret_cast<size_t>(pResolvedToken->hField));
 
-            if (pFieldInfo->fieldFlags & CORINFO_FLG_FIELD_INITCLASS)
+            if (isStaticReadOnlyInited)
             {
-                indirFlags |= GTF_IND_INITCLASS;
-            }
-            if (isStaticReadOnlyInitedRef)
-            {
-                indirFlags |= (GTF_IND_INVARIANT | GTF_IND_NONNULL);
```

Comment:
`GTF_IND_NONNULL` is likely a bug here, we didn't check if its value is null anywhere here.